src/util/color/color.h

```diff
@@ -0,0 +1,55 @@
+#ifndef PREDEFINEDCOLORSET_H
+#define PREDEFINEDCOLORSET_H
+
+#include <QColor>
+#include <QHash>
+
+#include "predefinedcolorsset.h"
+#include "util/math.h"
+
+namespace Color {
+    static const PredefinedColorsSet predefinedColorSet = PredefinedColorsSet();
+
+    // algorithm by http://www.nbdtech.com/Blog/archive/2008/04/27/Calculating-the-Perceived-Brightness-of-a-Color.aspx
+    // NOTE(Swiftb0y): please suggest if I should you use other methods
+    // (like the W3C algorithm) or if this approach is to to performance hungry
+    // NOTE: the author did not take alpha transparency into account!
+    static inline int brightness(int red, int green, int blue) {
```

Comment:
I think we use it where we have the hsv representation of the colour anyway where value is the brightness.

It finally depends on the display device. We can't accurate calculate the brightness from an RGB, since it depends on the device showing the values. 
If we want to make a "serious" implementation, we have to move to the x,y,L representation of a colour. This is used two show exact the same color at two different displays. To convert this to RGB, we need to have access to the monitor calibration data.

Reply:
I don't think it makes a difference internally if we use HSV or RGB, but most controllers deal with rgb components (eg on the launchpad mk2 you can [set the color of a pad via a sysex message](https://customer.novationmusic.com/sites/customer/files/novation/downloads/10529/launchpad-mk2-programmers-reference-guide-v1-02.pdf) (page 12)). So I think it would be easier if the mapping creator had the RGB color (-components) exposed directly.